mpyq.py

```diff
@@ -252,29 +252,28 @@ def decompress(data):
 
             return file_data
 
-    def extract(self):
+    def extract(self, *filenames):
         """Extract all the files inside the MPQ archive in memory."""
-        if self.files:
-            return dict((f, self.read_file(f)) for f in self.files)
+        files = filenames if len(filenames)>0 else self.files
+        if files:
+            return dict((f, self.read_file(f)) for f in files)
         else:
             raise RuntimeError("Can't extract whole archive without listfile.")
 
-    def extract_to_disk(self):
+    def extract_to_disk(self, *filepaths, **target_dir):
         """Extract all files and write them to disk."""
+        target_dir = target_dir['target_dir'] if 'target_dir' in target_dir else os.getcwd() # Need to write this way for 2.7 :(
```

Comment:
The way I wanted to write it is not allowed in Python2.7:

```
Python 2.7.18 |Anaconda, Inc.| (default, Jun  4 2021, 14:47:46)
[GCC 7.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> def function(self, *args, target_dir='default'):
  File "<stdin>", line 1
    def function(self, *args, target_dir='default'):
```

Hence the comment. Similarly the temp file testing is a bit more awkward due to 2.7 limitations.

Reply:
That's because varargs parameters must become after normal parameters:

```
Python 2.7.18 (default, Mar 15 2021, 14:29:03) 
[GCC 10.2.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> def function(self, target_dir='default', *args):
...     print "All good!"
... 
```

Reply:
Ah, but then that consumes one of the varargs. Right, sorry for the confusion.